packages/server/api/controllers/variants.controller.js

```diff
@@ -0,0 +1,24 @@
+const knex = require('../../config/db');
+const HttpError = require('../lib/utils/http-error');
+
+const getVariantsByProductId = async (productId) => {
+  if (!productId) {
+    throw new HttpError('Id should be a number', 400);
+  }
+
+  try {
+    const variants = await knex('variants')
+      .select('*')
+      .where('product_id', `${productId}`);
+    if (variants.length === 0) {
+      throw new HttpError(`incorrect entry with the id of ${productId}`, 404);
+    }
+    return variants;
+  } catch (error) {
+    return error.message;
```

Comment:
By having this catch here, you remove the point of using HttpError, since it will catch that an unwrap it again. 

You could check if the `error` is a instance of `HttpError` and act based on that.